Tests/SonarQube.TeamBuild.PostProcessor.Tests/MSBuildPostProcessorTests.cs

```diff
@@ -254,6 +256,92 @@ public void PostProc_ValidArgsPassedThrough()
             context.TargetsUninstaller.Verify(m => m.UninstallTargets(context.Logger));
         }
 
+        [TestMethod]
+        public void PostProc_WhenSettingInFileButNoCommandLineArg_Fail()
+        {
+            // Arrange
+            var context = new PostProcTestContext(TestContext);
+            context.Config.AdditionalConfig = new List<ConfigSetting> { new ConfigSetting { Id = ConfigSettingsExtensions.IsUsingCommandLineCredentialsKey } };
+
+            // Act
+            var success = Execute(context, args: new string[0]);
+
+            // Assert
+            Assert.IsFalse(success);
+            context.Logger.AssertErrorLogged("Credentials must be passed in both begin and end steps or not at all");
```

Comment:
Might be better to either directly use the message property from the resource file, or define this message as a constant in this class to avoid repeating it in every test.

Reply:
I'll go for the const because I quite like failing tests when I change a value in the resx.